src/Compilers/VisualBasic/Portable/Binding/Binder_Statements.vb

```diff
@@ -2279,12 +2279,15 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                         hasErrors = True
                     Else
                         Dim useSiteDiagnostics As HashSet(Of DiagnosticInfo) = Nothing
+                        Dim accessThroughType = GetAccessThroughType(actualEventAccess.ReceiverOpt)
+                        If Not (Me.IsAccessible(method, useSiteDiagnostics, accessThroughType) OrElse
+                            Not IsAccessible(eventSymbol, useSiteDiagnostics, accessThroughType)) Then
```

Comment:
> Would it not be an issue if they are referencing events from outside of VB?

Possibly. If that is a scenario that matters though why didn't you add tests for it?